src/views/GoTV/useTwitchIntegration.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C)  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useState, useEffect } from "react";
+import { usePreference } from "preferences";
+
+// Client ID will need to be the OGS client ID, probably from environment variable?
+// Redirect URI also needs to match the allowed redirect URIs in the console at dev.twitch.tv
+const TWITCH_CLIENT_ID = "z00yx3qzdzkh10c6nxku6tqm1pxrgv"; // cspell:disable-line
+const REDIRECT_URI = "http://localhost:8080/settings/gotv";
```

Comment:
Maybe construct this with `window.location.protocol` and `window.location.host` so it works on beta/localhost/production?

Reply:
Good point, I updated that